Emby.Server.Implementations/Session/WebSocketController.cs

```diff
@@ -103,5 +103,25 @@ public void Dispose()
 
             _disposed = true;
         }
+
+        public async ValueTask DisposeAsync()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            foreach (var socket in _sockets)
+            {
+                socket.Closed -= OnConnectionClosed;
+
+                if (socket is IAsyncDisposable disposableAsync)
```

Comment:
No need for this check if IWebSocketConnection implements IAsyncDisposable

Reply:
I was using that to cast the socket to `IAsyncDisposable` so I could call `DisposeAsync()` on the socket. I think because `WebSocketController` doesn't inherit `WebSocketConnection` I can't use `socket.DisposeAsync()` without the cast. I could instead do `await ((IAsyncDisposable)socket).DisposeAsync().ConfigureAwait(false)` if the if statement seems nonsensical.

Reply:
But `socket` is `IWebSocketConnection`

Reply:
Got `IAsyncDisposable` added to the interface and it's working great! Sorry for the mistake on that last review. 

I went ahead and made the sockets dispose in `Dispose()` in `WebSocketController`. I also added the private boolean `_disposed` in `WebSocketConnection` to prevent `Disposed()` from getting called again.